portal-api/src/v1/gef/models/facilities.js

```diff
@@ -0,0 +1,45 @@
+class Facility {
+  constructor(req) {
+    if (req.applicationId) {
+      // new application
+      this.applicationId = req.applicationId ? req.applicationId : null;
+      this.type = req.type !== null ? Number(req.type) : null; // 0 Cash | 1 Contingent
+      this.hasBeenIssued = null;
+      this.name = null;
+      this.startOnDayOfNotice = null;
+      this.coverStartDate = null;
+      this.coverEndDate = null;
+      this.monthsOfCover = null;
+      this.details = null;
+      this.detailsOther = null;
+      this.currency = null;
+      this.value = null;
+      this.coverPercentage = null;
+      this.interestPercentage = null;
+      this.paymentType = null;
+      this.createdAt = Date.now();
+      this.updatedAt = null;
+    } else {
+      // update application
+      this.hasBeenIssued = req.hasBeenIssued !== null ? Boolean(req.hasBeenIssued) : null;
+      this.name = req.name ? String(req.name) : null;
+      this.startOnDayOfNotice = req.startOnDayOfNotice !== null ? Boolean(req.startOnDayOfNotice) : null;
+      this.coverStartDate = req.coverStartDate ? new Date(req.coverStartDate) : null;
+      this.coverEndDate = req.coverStartDate ? new Date(req.coverStartDate) : null;
+      this.monthsOfCover = req.monthsOfCover ? Number(req.monthsOfCover) : null;
+      this.details = req.details ? req.details : null;
+      this.detailsOther = req.detailsOther ? String(req.detailsOther) : null;
+      this.currency = req.currency ? String(req.currency) : null;
+      this.value = req.value !== null ? Number(req.value) : null;
+      this.coverPercentage = req.coverPercentage !== null ? Number(req.coverPercentage) : null;
+      this.interestPercentage = req.interestPercentage !== null ? Number(req.interestPercentage) : null;
+      // 0 - In advance (quarterly) | 1 - In Arrears (quarterly)
+      this.paymentType = req.paymentType !== null ? Number(req.paymentType) : null;
+      this.updatedAt = Date.now();
+    }
+  }
+}
```

Comment:
Thinking ahead for mapping GEF deals / facilities - in a GEF deal/application, do we have a dealType ? e.g:

```
{
   ...application/deal,
   dealType: 'gef'
}
```

For facilities - in Portal we have `facilityType` as 'bond' or 'loan'. I can see there is `facility.type` of 0 or 1 for GEF facility.

Is there are benefit to having 0 or 1 values / what's the thinking behind it? Maybe coud just have it as 'Cash' or 'Contingent'?  

i'm just thinking of future TFM integration and mappings, particularly around reports/filtering.

Reply:
Good point. And if we need Bonds, Loans & Gef deals in a single list on a dashboard, won't we need them all in the same collection so mongo can do the sorting & filtering? So using deals collection instead on gef-application? Which also means aligning the fields used in GraphQL. 
Or maybe we need a new collection which contains index of deals & gef-applications and only contains fields required by dashboard?

Reply:
Seems that one collection of deals/applications, for GEF and Portal - and similarily a single collection for all facilities, would be good way to go. I think splitting out would have more complications - e.g if dashboard wants to add or change a filter in the future, could be a pain.

If all in the same collection - the data models would be different, but I think that's okay as we can do mappings in the graphql queries/filtering, based on `dealType` or `facilityType`. In TFM there's already some differnet mappings happening depending on the facilityType of bond/loan.

But I think either way it'd be good to have `dealType` and `facilityType` aligned; otherwise would be confusing having some facilities as "type 0" or "type 1" if the other facilities are not numbered like this

Reply:
@mwyld We'll need to store the GEF application in the same collection. As there'll be additional work around graphql to get this working it may be better to raise a separate ticket to do this so it doesn't hold Rich up.

@mwyld Can you align facilityType with Portal values in this ticket before submitting? i.e. have the value as gef-loan & gef-bond or whatever the gef equivalent is.

We'll need to raise the additional work for the collections with Gavin.

Reply:
I agree lets raise a separate ticket on this and have a chat about amending it, so it doesn't block Richy

Reply:
Just realised we also need to have an `associatedDealId` in each facility